libnmstate/state.py

```diff
@@ -246,3 +438,28 @@ def dict_update(origin_data, to_merge_data):
         else:
             origin_data[key] = val
     return origin_data
+
+
+def _is_route_match(route, other):
+    if Route.DESTINATION in route and \
```

Comment:
Note sure how @EdDev sees this but you could just iterate over all the possible keys, for example

```
for key in Route.DESTINATION, Route.NEXT_HOP_INTERFACE:
    if key in route and route[key] != other[key]:
        return False
```
Not sure if there is a better name for `key`.

Reply:
Changed to:

```python
def _is_route_match(route, other):
    for prop_name in (Route.DESTINATION, Route.NEXT_HOP_INTERFACE,
                      Route.NEXT_HOP_ADDRESS, Route.METRIC,
                      Route.TABLE_ID):
        if prop_name in Route and route[prop_name] != other[prop_name]:
            return False
    return True
```

Thanks for the hints.